src/Utils/hooks/useTracking.ts

```diff
@@ -0,0 +1,24 @@
+import { SystemContext } from "Artsy"
+import * as AnalyticsSchema from "Artsy/Analytics/Schema"
+import { useContext, useEffect } from "react"
+import { TrackingProp } from "react-tracking"
+
+export function useTracking(
+  newTrackingInstance?: TrackingProp
+): {
+  tracking: TrackingProp
+  AnalyticsSchema: typeof AnalyticsSchema
+} {
+  const { tracking, setTracking } = useContext(SystemContext)
+
+  useEffect(() => {
+    if (newTrackingInstance) {
+      setTracking(newTrackingInstance)
+    }
+  }, [])
+
+  return {
+    tracking,
+    AnalyticsSchema,
```

Comment:
I'm a bit concerned about passing the `AnalyticsSchema` as a prop here.  Might seem convenient, but it's only an import away, and the TS language server can do auto importing so it's even easier than destructuring the return value of `useTracking` if you lean into intellisense.

Also, importing `AnalyticsSchema` will still be required when wrapping things with `track` so having multiple ways to get at those values might create confusion. Especially when the destructuring approach implies dependency injection, which is not happening.

Personally I'd rather see a single named export for AnalyticsSchema (rather than the many options we have right now) which makes the auto import trivial, while also making it so there's one standard way to access analytics values 💆

Reply:
The main thought is convenience. There are multiple ways we handle tracking in Reaction, and this just makes things a little quicker without much possibility of causing harm. We're already returning `tracking` from `useTracking`, why not also pass along the schema. I think the chance of confusion is fairly small.

Reply:
I think your idea of convenience is different from my idea of convenience :)